src/store/store.ts

```diff
@@ -186,4 +187,22 @@ export class Store implements Cache {
   ): void {
     writeFragment(this, dataFragment, data, variables);
   }
+
+  hydrateData(data: object, adapter: StorageAdapter) {
+    InMemoryData.storage.current = adapter;
+    InMemoryData.initDataState(this.data, 0);
+    for (const key in data) {
+      const entityKey = key.slice(2, key.indexOf('.'));
+      const fieldKey = key.slice(key.indexOf('.') + 1);
+      switch (key.charCodeAt(0)) {
+        case 108:
+          InMemoryData.writeLink(entityKey, fieldKey, data[key], true);
```

Comment:
kinda on the ropes since we are deleting with `undefined` what if a certain adapter writes `undefined` we could start writing this to cache which isn't ideal maybe we should guard? Or opt for a full `remove` method?